### PR TITLE
Feature/support external data

### PR DESCRIPTION
**Pull request by KateSakharova**
- added samplesheet support for multiple runs

### CODE REVIEW

File: assets/schema_input.json [hunk 1]
```diff
@@ -29,8 +34,18 @@
...
+                "errorMessage": "library_layout should be only 'single' or 'paired' value"
+            },
+            "library_strategy": {
+                "type": "string",
+                "enum": ["metagenomic", "metatranscriptomic", "genomic", "transcriptomic", "other"],
+                "errorMessage": "library strategy should be only value from list: 'metagenomic', 'metatranscriptomic', 'genomic', 'transcriptomic', 'other'"
             }
         },
-        "required": ["sample", "fastq_1"]
+        "required": ["study_accession","reads_accession", "fastq_1", "library_layout", "library_strategy"]
```
Comment (Ge94) [inline assets/schema_input.json:49]: Just to confirm, these are all fields that will be received from prefect?

File: assets/schema_input.json [hunk 2]
```diff
@@ -29,8 +34,18 @@
...
+            },
+            "library_layout": {
+                "type": "string",
+                "enum": ["single", "paired"],
+                "errorMessage": "library_layout should be only 'single' or 'paired' value"
+            },
+            "library_strategy": {
+                "type": "string",
+                "enum": ["metagenomic", "metatranscriptomic", "genomic", "transcriptomic", "other"],
+                "errorMessage": "library strategy should be only value from list: 'metagenomic', 'metatranscriptomic', 'genomic', 'transcriptomic', 'other'"
```
Comment (Ge94) [inline assets/schema_input.json:46]: How will the "other" case be handled from prefect? e.g. in the case metadata are wrong and we realise the data are metaG/metaT and so on, is prefect going to default everything to "other" or is a manual input needed from us? In the latter case, what does "other" cover?

File: nextflow_schema.json [hunk 3]
```diff
@@ -10,14 +10,27 @@
...
+            "oneOf": [
+                {
+                    "description": "Required fields when samplesheet is not provided.",
+                    "required": [
+                      "study_accession",
+                      "reads_accession",
+                      "blast_reference_genomes_folder",
+                      "bwamem2_reference_genomes_folder",
+                      "outdir",
+                      "human_phix_blast_index_name",
```
Comment (Ge94) [inline nextflow_schema.json:22]: Would these two index names be required also in the other case?

File: assets/schema_input.json (hunk 1 above)
Comment (Ge94) [reply assets/schema_input.json:49]: To clarify, I see these will be overridden in the miassembler workflow when a samplesheet is used. So, are these still required fields?

**Review by Ge94 — COMMENTED** [review]
Nice work! I just left a couple clarification comments here and there

File: assets/schema_input.json (hunk 1 above)
Comment (mberacochea) [reply assets/schema_input.json:49]: The idea is either to submit --study and --run-accession OR to use a samplesheet. The values from the samplesheet have the highest priority.

File: assets/schema_input.json (hunk 2 above)
Comment (KateSakharova) [reply assets/schema_input.json:46]: I always thought OTHER is ENA notation/option

File: nextflow_schema.json (hunk 3 above)
Comment (KateSakharova) [reply nextflow_schema.json]: Yes, I think you are right

File: assets/schema_input.json (hunk 2 above)
Comment (Ge94) [reply assets/schema_input.json:46]: Ah yes, it does. I was wondering whether prefect allows for "other" values, then this is great

**Review by Ge94 — APPROVED** [review]

**Review by Ge94 — APPROVED** [review]

**Pull request merged**